server/src/main/java/org/elasticsearch/search/aggregations/support/CoreValuesSourceType.java

```diff
@@ -224,7 +233,110 @@ public ValuesSource getField(FieldContext fieldContext, AggregationScript.LeafFa
         public ValuesSource replaceMissing(ValuesSource valuesSource, Object rawMissing, DocValueFormat docValueFormat, LongSupplier now) {
             throw new IllegalArgumentException("Can't apply missing values on a " + valuesSource.getClass());
         }
-    };
+    },
+    // TODO: Ordinal Numbering sync with types from master
+    IP(EquivalenceType.STRING) {
```

Comment:
The three new `CoreValuesSourceType`s exist mosty to hold formatters, functionality migrated from `ValueType`.  The goal is for `ValueSourceType` to hold all the information the aggregation needs to parse values, rather than having it spread between two or more classes.